### ISSUE TITLE
Is encryption supposed to work already?

### ISSUE THREAD

**Issue by igrokit** [joaomgcd/node-red-contrib-join-joaoapps#2]

**Comment by joaomgcd** [comment]
Sorry, not yet.

**Comment by shanness** [comment]
Hmm, damn, is there any way around that in node-red?  I've already got all my devices talking encrypted and really don't want to turn that off if possible.  P.S. Thanks for this, it looks very powerful and useful!  Loving all your apps!

**Comment by joaomgcd** [comment]
Not right now, sorry! I will have to check if there's a way to implement encryption on it.

**Comment by shanness** [comment]
What about the option to pass the unencrypted data from chrome/android/whatever to the node-red port?  Wouldn't that work?  i.e. change the chrome extension to pass unencrypted data through?  This seems safe as it's usually localhost?  And I assume all the join endpoints that can pass to node-red already decrypt the data, so sounds easy?

**Comment by joaomgcd** [comment]
Yes, I actually considered that option but I'm afraid people will say that I'm doing unencrypted messaging even though encryption is enabled and think I'm doing it on purpose to steal their data or something...

**Comment by shanness** [comment]
Fair enough, I'll keep an eye on this project then for you adding decryption.   For now I'm just creating node-red http input nodes and hitting them directly from tasker.

**Comment by shanness** [comment]
Hey, was just about to try adding the decryption code from the chrome app to this, then saw the node red pass through in the chrome app settings..  This handles it, decrypting it for me.  So for people running node red on a the same lan as an always on desktop, just set the port (from the node-red join config) and host in advanced chrome app settings and it all works.

Putting this here for others that might not realise this simple work around (wasn't at all clear to me in the any docs and [this site](
https://notenoughtech.com/home-automation/nodered-home-automation/how-to-use-join-app-in-nodered/
) makes it sound like it won't work (big red box about needing to disable encryption directly under the chrome app setup).

Basically you can keep using encryption, just push commands to chrome instead of node-red, and they'll get forward unencrypted!

**Comment by shanness** [comment]
Doh, silly me, was tired when I did that.  Of course that only solves half the problem, receiving.  I suppose I'll have to use the join web hits to send stuff the other way :(  But looking at the JoinChrome code, it looks like it would be pretty easy for you to copy the encryption stuff in util over to this?  Would be great to have it working properly.

**Comment by joaomgcd** [comment]
This is now implemented :)

**Comment by shanness** [comment]
Hey @joaomgcd , I installed 1.2, and get these errors.
```
[node-red-contrib-join-joaoapps/join-receive-message] SyntaxError: Unexpected token { (line:6)
[node-red-contrib-join-joaoapps/join-message] SyntaxError: Unexpected token { (line:5)
```
Went back to `npm install node-red-contrib-join-joaoapps@1.1.3` and it went away.

**Comment by shanness** [comment]
Was complex to fix, a few posts said I needed node 12.13 to use it, and I was running node-red on v8.  Anyhow after a very painful upgrade, all seems to be working now, thanks :)

You might wanna put a warning up for others if 1.2 needs node 12.13.  https://joaoapps.com/join/node-red/ says v8

**Comment by joaomgcd** [comment]
Thank you very much for the info. Updated the page 👍
